### PR TITLE
feat(savers): more deposit/withdraw safety

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- Brings a similar missing UTXO account address safety added as part of https://github.com/shapeshift/web/pull/3643 CR, this time for deposits
- Makes sure we throw actual sign/broadcast errors for non-UTXOs
- Tightens the deposit async flow similarly to what was done in https://github.com/shapeshift/web/pull/3643

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

None, isolated to savers

### Engineering

- This looks like code

### Operations

- None

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3583** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3583" target="_blank">Graphite</a> 
    * **PR #3587** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3587" target="_blank">Graphite</a> 
      * **PR #3600** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3600" target="_blank">Graphite</a> 
        * **PR #3598** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3598" target="_blank">Graphite</a> 
          * **PR #3612** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3612" target="_blank">Graphite</a> 
            * **PR #3614** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3614" target="_blank">Graphite</a> 
              * **PR #3626** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3626" target="_blank">Graphite</a> 
                * **PR #3613** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3613" target="_blank">Graphite</a> 
                  * **PR #3629** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3629" target="_blank">Graphite</a> 
                    * **PR #3630** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3630" target="_blank">Graphite</a> 
                      * **PR #3640** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3640" target="_blank">Graphite</a> 
                        * **PR #3644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3644" target="_blank">Graphite</a> 
                          * **PR #3643** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3643" target="_blank">Graphite</a> 
                            * **PR #3664** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3664" target="_blank">Graphite</a>  👈
                            * **PR #3635** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3635" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3664?utm_source=stack-comment).

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx [hunk 1]
```diff
@@ -398,27 +398,27 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
     const txId = await handleSend({
       sendInput: withdrawInput,
       wallet: walletState.wallet,
-    }).catch(async () => {
+    }).catch(async e => {
+      if (!isUtxoChainId(chainId)) throw e
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx:402]: If this fails, this is an actual error, no need to intent a pre-withdraw Tx

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx [hunk 2]
```diff
@@ -218,37 +228,38 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
...
-        wallet: walletState.wallet,
-      })
-    } catch (e) {
+    // 1. Try to deposit from the originally deposited from / highest UTXO balance address
+    // If this is enough, no other Tx is needed
+    const txId = await handleSend({
+      sendInput: depositInput,
+      wallet: walletState.wallet,
+    }).catch(async e => {
+      if (!isUtxoChainId(chainId)) throw e
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx:243]: If this fails, this is an actual sign/broadcast error, not an UTXO coinselect one no need to intent a pre-withdraw Tx

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx [hunk 3]
```diff
@@ -218,37 +228,38 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
 
     const depositInput = await getDepositInput()
     if (!depositInput) throw new Error('Error building send input')
-
-    let txId: string
+    if (!walletState?.wallet) throw new Error('Wallet is required')
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx:231]:
```suggestion
```

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**